lib/much-result.rb

```diff
@@ -193,11 +193,20 @@ def to_much_result(backtrace: caller, **kargs)
     self.set(**kargs)
   end
 
+  def attributes
+    @data.to_h.reject { |key, _| key.to_s.start_with?("much_result_") }
+  end
+
+  def attribute_names
+    attributes.keys
+  end
+
```

Comment:
I'd like to juxtapose these methods right after the `def set(**kargs)` method above please since they are part of the same workflow (setting and introspecting on arbitrary attributes).